docs/opc-event-output.md

```diff
@@ -0,0 +1,199 @@
+```json
+{
+  "body": [
+    {
+      "MessageId": "43",
+      "MessageType": "ua-data",
+      "PublisherId": "SIMPLE-EVENTS",
+      "DataSetWriterGroup": "SIMPLE-EVENTS",
+      "Messages": [
+        {
+          "DataSetWriterId": "SIMPLE-EVENTS",
+          "MetaDataVersion": {
+            "MajorVersion": 1,
+            "MinorVersion": 0
+          },
+          "Payload": {
+            "SimpleEvents": {
+              "Value": {
+                "EventId": "+6CQjN1eqkO6+yHJnxMz5w==",
+                "EventType": "http://microsoft.com/Opc/OpcPlc/SimpleEvents#i=14",
+                "Message": "The system cycle '59' has started.",
+                "ReceiveTime": "2021-06-21T12:38:55.5814091Z",
+                "Severity": 1,
+                "SourceName": "System",
+                "SourceNode": "i=2253",
+                "Time": "2021-06-21T12:38:55.5814078Z"
+              },
+              "SourceTimestamp": "2021-06-21T12:38:55.5814091Z"
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "enqueuedTime": "Mon Jun 21 2021 14:39:02 GMT+0200 (Central European Summer Time)",
+  "properties": {
+    "$$ContentType": "application/x-network-message-json-v1",
+    "iothub-message-schema": "application/ua+json",
+    "$$ContentEncoding": "utf-8"
+  }
+}
+```
+A few things to note here. Here the event in the payload is named "SimpleEvents" which has been configured to by setting the DisplayName property of the events configuration. Also note that all fields+values reside under the Value key.
+
+The format produced here does not contain enough information to decode the JSON properly. If you need this there is a command-line switch in the OPC Publisher called "reversibleencoding", or "re" which can be set to either true or false. The default value is false. If you enable this setting the output will look like this:
+```json
+{
+  "body": [
+    {
+      "MessageId": "15",
+      "MessageType": "ua-data",
+      "PublisherId": "SIMPLE-EVENTS",
+      "DataSetWriterGroup": "SIMPLE-EVENTS",
+      "Messages": [
+        {
+          "DataSetWriterId": "SIMPLE-EVENTS",
+          "MetaDataVersion": {
+            "MajorVersion": 1,
+            "MinorVersion": 0
+          },
+          "Payload": {
+            "SimpleEvents": {
+              "Value": {
+                "Type": "ExtensionObject",
+                "Body": {
+                  "TypeId": "s=EncodeableDictionary",
+                  "Encoding": "Json",
+                  "Body": {
+                    "_Count": 8,
+                    "_Keys": [
+                      "EventId",
+                      "EventType",
+                      "Message",
+                      "ReceiveTime",
+                      "Severity",
+                      "SourceName",
+                      "SourceNode",
+                      "Time"
+                    ],
+                    "EventId": {
+                      "Type": "ByteString",
```

Comment:
strange, builtin types should be encoded as number

Reply:
Should this be encoded as a number?
![image](https://user-images.githubusercontent.com/5628905/122891365-c158ef00-d344-11eb-81c4-f5935ebf53a2.png)

Reply:
Or do you mean some other field?

Reply:
The type

Reply:
example for encoding of a double:
```json
 "#0": {
        "Value": {
          "Type": 11,
          "Body": 6628948.417515845
        },

Reply:
ah, ok. I see.